components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/definitions/AsyncApiParserTest.java

```diff
@@ -0,0 +1,98 @@
+ *  limitations under the License.
+ */
+package org.wso2.carbon.apimgt.impl.definitions;
+
+import org.apache.commons.io.IOUtils;
+import org.json.JSONException;
+import org.junit.Assert;
+import org.junit.Test;
+import org.wso2.carbon.apimgt.api.model.API;
+import org.wso2.carbon.apimgt.api.model.APIIdentifier;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+public class AsyncApiParserTest {
+
+    @Test
+    public void testGenerateAsyncAPIDefinition() throws Exception {
+        APIIdentifier identifier = new APIIdentifier("admin", "HelloServer", "1.0");
+        API api = new API(identifier);
+        api.setType("WS");
+        api.setContext("/hello/1.0");
+        api.setTransports("ws,wss");
+        api.setEndpointConfig("{'endpoint_type':'http','sandbox_endpoints':{'url':'wss://echo.websocket.org:443'}," +
+                "'production_endpoints':{'url':'wss://echo.websocket.org:443'}}");
+        AsyncApiParser asyncApiParser = new AsyncApiParser();
+        String asyncAPIDefinition = asyncApiParser.generateAsyncAPIDefinition(api);
+        Assert.assertNotNull(asyncAPIDefinition);
+    }
+
+    @Test
+    public void testGetAsyncAPIDefinitionForStore() throws Exception {
+        Map<String, String> hostsWithSchemes = new HashMap<>();
+        hostsWithSchemes.put("ws", "ws://localhost:9099");
+        hostsWithSchemes.put("wss", "wss://localhost:8099");
+        APIIdentifier identifier = new APIIdentifier("admin", "HelloServer2", "1.0");
+        API api = new API(identifier);
+        api.setType("WS");
+        api.setContext("/hello2/1.0");
+        api.setTransports("ws,wss");
+        api.setEndpointConfig("{'endpoint_type':'http','sandbox_endpoints':{'url':'wss://echo.websocket.org:443'}," +
+                "'production_endpoints':{'url':'wss://echo.websocket.org:443'}}");
+        AsyncApiParser asyncApiParser = new AsyncApiParser();
+        String asyncAPIDefinition = IOUtils.toString(
+                getClass().getClassLoader().getResourceAsStream("definitions" + File.separator + "asyncAPI" +
+                        File.separator + "sampleWebSocket.json"),
+                "UTF-8");
+        api.setAsyncApiDefinition(asyncAPIDefinition);
+        String definitionForStore = asyncApiParser.getAsyncApiDefinitionForStore(api, asyncAPIDefinition, hostsWithSchemes);
+        Assert.assertNotNull(definitionForStore);
+    }
+
+    @Test
+    public void testGenerateAsyncAPIDefinitionFail() throws Exception {
+        APIIdentifier identifier = new APIIdentifier("admin", "HelloServer", "1.0");
+        API api = new API(identifier);
+        AsyncApiParser asyncApiParser = new AsyncApiParser();
+        try {
+            asyncApiParser.generateAsyncAPIDefinition(api);
+        } catch (JSONException e) {
+            Assert.assertNotNull(e);
+        }
+    }
+
+    @Test
+    public void testGetAsyncAPIDefinitionForStoreFail() throws Exception {
```

Comment:
can there be a case where async api will be generated for wrong format of ws url ?

Reply:
No. This is to create a mock scenario of WS URLs. There are checks to validate the URL when adding to the API object in actual flows.